node/core/candidate-selection/src/lib.rs

```diff
@@ -163,7 +170,7 @@ impl CandidateSelectionJob {
 		}
 	}
 
-	async fn run_loop(&mut self, span: &jaeger::Span) -> Result<(), Error> {
+	async fn run_loop(&mut self, span: &jaeger::JaegerSpan) -> Result<(), Error> {
```

Comment:
```suggestion
	async fn run_loop(&mut self, span: impl Into<jaeger::JaegerSpan>) -> Result<(), Error> {
```

Writing this as something which can be converted into a `JaegerSpan` means that at line 476, you can just hand it `None`, instead of explicitly constructing a `jaeger::JaegerSpan::Disabled`.

Note that the suggested change gets rid of the reference as well. Moving the span like that means that it drops when `run_loop` exits, not when its creating scope exits, but as the `run_loop` call is the final line of that enclosing scope, this should be acceptable. Still, it implies updating the call at line 152 as well.

Reply:
Not keen on introducing this change. It trades additional complexity for hiding an enum variant, which makes it very explicit what's happening.